src/odemis/acq/stream/_sync.py

```diff
@@ -2324,6 +2324,38 @@ def _adjustHardwareSettings(self):
         self._emitter.scale.value = cscale
         return self._dwellTime.value
 
+    def _get_next_rectangle(self, rep: Tuple[int, int], acq_num: int, px_time: float,
+                            leech_np: List[Optional[int]]) -> Tuple[int, int]:
+        """
+        Get the next rectangle to scan, based on the leeches and the live update period.
+        :param rep: total number of pixels to scan (y, x)
+        :param acq_num: number of pixels scanned so far
+        :param px_time: time to acquire one pixel
+        :param leech_np: number of pixels to scan before the next run of each leech.
+        If None, the leech doesn't need to run.
+        :return: size of the rectangle to scan (y, x)
+        """
+        tot_num = rep[0] * rep[1]
+        # The very max number of pixels to acquire
+        npixels2scan = min([np for np in leech_np if np is not None] + [tot_num - acq_num])
```

Comment:
I don't understand why you use `min` if it is supposed to be the max number of pixels to acquire